Source/missiles.cpp

```diff
@@ -267,8 +269,10 @@ bool MonsterMHit(int pnum, int m, int mindam, int maxdam, int dist, missile_id t
 
 	if (!shift)
 		dam <<= 6;
-	if (resist)
-		dam >>= 2;
+
+	assert(dam < (0x7FFFFFFF >> 6));

```

Comment:
Probably use `std::numeric_limits<int>::max()`

Reply:
agreed, I didn't realize that was a thing at the time 🙂